### PR TITLE
WAFV2 : list-web-acls

### PR DESCRIPTION
**Pull request by v-dev**
**Background:**
Related to #3971

Begin adding support for the wafv2 services, starting with the `list-web-acls` API. Currently just returns 2 preloaded mock WebACLs.

First PR in this project. Appreciate the guidance.

**Sample Output:**
```
$ aws --endpoint-url=http://localhost:5000 wafv2 list-web-acls --scope REGIONAL
{
    "WebACLs": [
        {
            "Name": "first-mock-webacl-ad54c7ad-8a89-450f-9673-9c1571c6acc3",
            "Id": "edcf8c35-3615-4b38-a57f-96efb0f67945",
            "Description": "Mock WebACL named first-mock-webacl-ad54c7ad-8a89-450f-9673-9c1571c6acc3",
            "ARN": "arn:aws:wafv2:us-east-1:123456789012:regional/webacl/first-mock-webacl-ad54c7ad-8a89-450f-9673-9c1571c6acc3/fecf928a-4ae5-404e-9791-91ff0d299ffe"
        },
        {
            "Name": "second-mock-webacl-bfbb5048-1037-4fd5-bebc-f1e00f6c1717",
            "Id": "e02e0290-5d37-43d1-a294-7e1339d0530d",
            "Description": "Mock WebACL named second-mock-webacl-bfbb5048-1037-4fd5-bebc-f1e00f6c1717",
            "ARN": "arn:aws:wafv2:us-east-1:123456789012:regional/webacl/second-mock-webacl-bfbb5048-1037-4fd5-bebc-f1e00f6c1717/5f79a574-a240-4a78-9fad-f14a263c6971"
        }
    ]
}
```

### CODE REVIEW

File: moto/wafv2/models.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        """
+        The name must have 1-128 characters. Valid characters: A-Z, a-z, 0-9, - (hyphen), and _ (underscore).
+        """
+        self.Name = "{0}-{1}".format(name, uuid.uuid4())[0:128]
+        self.ARN = make_webacl_arn(US_EAST_1_REGION, self.Name)
+        self.Capacity = 3
+        self.Description = "Mock WebACL named {0}".format(self.Name)
+        self.VisibilityConfig = VisibilityConfig(self.Name)
+        """
+        TODO: create classes for the below fields:
```
Comment (v-dev) [inline moto/wafv2/models.py:40]: There are many [other fields in a WebACL](https://docs.aws.amazon.com/waf/latest/APIReference/API_WebACL.html), and those, in turn, often have many other fields, too. Should I put them all in this `models.py` file, or should I create a `models` package with the other classes?

Comment (bblommers) [reply moto/wafv2/models.py:40]: It's fine to have them all in the same file.
For most services, if there's a model with many (nested) fields, we tend to just store them as a dictionary. Depending on how many of the fields are actually used, or whether they're simply stored for retrieval later

File: moto/wafv2/responses.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+class WAFV2Response(BaseResponse):
+    SERVICE_NAME = "wafv2"
+
+    @property
+    def wafv2_backend(self):
+        return wafv2_backends[self.region]
+
+
+class WafV2Handler(BaseResponse):
```
Comment (bblommers) [inline moto/wafv2/responses.py:17]: These two classes can be combined (`WAFV2Response`/`WavV2Handler`)

File: ubuntu_init_dev_setup.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/bash
+
+# from https://tutorials.technology/solved_errors/10-fatal-error-libxml-xpath_h-No-such-file-or-directory.html
+sudo apt-get install libxml2-dev libxslt1-dev python-dev
```
Comment (bblommers) [inline ubuntu_init_dev_setup.sh:4]: This shouldn't be in an separate file. Were you getting this error when running `make init`?

File: moto/wafv2/models.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    https://docs.aws.amazon.com/waf/latest/APIReference/API_VisibilityConfig.html
+    """
+
+    def __init__(self, metric_name):
+        self.SampledRequestsEnabled = True
+        self.CloudWatchMetricsEnabled = False
+        self.MetricName = metric_name
+
+
+wafv2_backends = {"global": WAFV2Backend(GLOBAL_REGION)}
```
Comment (bblommers) [inline moto/wafv2/models.py:66]: WAF is interesting, in that it combines both regional and global resources. As far as I know, it's the first service in Moto that does it.
The global/cloudfront resources are still persisted in `us-east-``, though. So the backends can still be created as every other service, having a separate backend per region. We'll just have to check for every call whether the scope was set to CLOUDFRONT or REGIONAL.

https://docs.aws.amazon.com/cli/latest/reference/wafv2/list-web-acls.html
>
> To work with CloudFront, you must also specify the Region US East (N. Virginia)

Comment (bblommers) [reply moto/wafv2/models.py:66]: The `get_available_regions` command does not return anything for `wafv2` at the moment, so let's use `waf-regional` for now. The list should be the same anyway.
```
wafv2_backends = {}
for region in Session().get_available_regions("waf-regional"):
    wafv2_backends[region] = WAFV2Backend(region)
for region in Session().get_available_regions("waf-regional", partition_name="aws-us-gov"):
    wafv2_backends[region] = WAFV2Backend(region)
for region in Session().get_available_regions("waf-regional", partition_name="aws-cn"):
    wafv2_backends[region] = WAFV2Backend(region)
```

File: moto/wafv2/utils.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import uuid
+from collections import namedtuple
+
+ARN = namedtuple("ARN", ["region", "account", "function_name", "version"])
+
+
+def make_arn(service, region, name):
+    return "arn:aws:wafv2:{0}:123456789012:regional/{1}/{2}/{3}".format(
```
Comment (bblommers) [inline moto/wafv2/utils.py:8]: The account ID is stored centrally:
`from moto.core import ACCOUNT_ID`

File: moto/wafv2/responses.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    def list_web_ac_ls(self):
+        wacl = wafv2_backends[GLOBAL_REGION]
```
Comment (bblommers) [inline moto/wafv2/responses.py:19]: `self.body` returns `{"Scope": "CLOUDFRONT"}` or  `{"Scope": "REGIONAL"}`, so we can use that to determine which backend to talk to.

File: moto/wafv2/models.py (hunk 1 above)
Comment (bblommers) [reply moto/wafv2/models.py:40]: Attribute-names should be pythonic btw (snake_case, so `self.name` and `self.visibility_config`)

**Comment by v-dev** [comment]
Thank you for all the feedback! I will incorporate and/or come back with more questions as I progress... which is more likely during the weekends.

File: ubuntu_init_dev_setup.sh (hunk 3 above)
Comment (v-dev) [reply ubuntu_init_dev_setup.sh:4]: It has been a while, but: I believe so, yes.

File: moto/wafv2/responses.py (hunk 2 above)
Comment (v-dev) [reply moto/wafv2/responses.py]: Combined into `WAFV2Response`

File: moto/wafv2/utils.py (hunk 5 above)
Comment (v-dev) [reply moto/wafv2/utils.py]: Good to know - updated... thanks!

File: moto/wafv2/models.py (hunk 4 above)
Comment (v-dev) [reply moto/wafv2/models.py]: Thank you. Updated.

File: moto/wafv2/responses.py (hunk 6 above)
Comment (v-dev) [reply moto/wafv2/responses.py]: Used `self.body` and added/updated tests.

**Comment by bblommers** [comment]
Closing in favour of #4076

**Pull request closed without merging**